src/jaxsim/rbda/contacts/relaxed_rigid.py

```diff
@@ -271,10 +272,10 @@ def compute_contact_forces(
             joint_force_references=joint_force_references,
         )
 
-        def _detect_contact(x: jtp.Array, y: jtp.Array, z: jtp.Array) -> jtp.Array:
+        def detect_contact(x: jtp.Array, y: jtp.Array, z: jtp.Array) -> jtp.Array:
             x, y, z = jax.tree.map(jnp.squeeze, (x, y, z))
 
-            n̂ = self.terrain.normal(x=x, y=y).squeeze()
+            n̂ = model.terrain.normal(x=x, y=y).squeeze()
```

Comment:
Do we still need `terrain` as a class attribute? Can it just be extracted from `model` ?

Reply:
Since it's part of the `ContactModel` base class, this is not the right place to remove it. As for `ContactModel.parameters` that are overriden by those in `data`.

I still have to wrap my head around those stored in the contact model itsels, as right now they act as default ones.